src/Interpreters/parseColumnsListForTableFunction.cpp

```diff
@@ -71,6 +74,34 @@ void validateDataType(const DataTypePtr & type, const DataTypeValidationSettings
                 "Set setting allow_experimental_variant_type = 1 in order to allow it", type->getName());
         }
     }
+
+    if (const auto * nullable_type = typeid_cast<const DataTypeNullable *>(type.get()))
```

Comment:
Ok, but:
- we have `IColumn::forEachSubcolumn`;
- we should add `IDataType::forEachChild`

(should not be named "subtype", because it has no algebraic meaning)